### DISCUSSION TITLE
ProviderRefBase and WidgetRef naming

### DISCUSSION THREAD

**Discussion by scopendo** [rrousselGit/riverpod#870] [General]
I noticed that with the 1.0 dev release that the ref passed to a provider's create function is now a ProviderRefBase, whereas a ConsumerWidget has WidgetRef.

I appreciate that these two need to be different but wondered if they could be named more consistently?  In particular, whether it might be possible (with refactor) for ProviderRefBase to be named ProviderRef?

Just thinking that little consistencies like that might help people to follow Riverpod more easily.

**Comment by rrousselGit** [comment]
In the master channel, ProviderRefBase was renamed as "Ref"

"ProviderRef" is taken by "Provider((ProviderRef ref)"

**Comment by TimWhiting** [comment]
Expounding on this. Each provider type has a different "Ref" type to allow different methods on for example AutoDispose providers, StateProviders etc.

`Provider((ProviderRef ref)`
`StateProvider((StateProviderRef ref)`
`FutureProvider((FutureProviderRef ref)`
...

**Comment by scopendo** [comment]
Thank you both. Makes sense and pleased that it'll be moving to Ref soon 👍
